jina/types/score/__init__.py

```diff
@@ -0,0 +1,53 @@
+from typing import Optional
+
+
+from ...excepts import BadNamedScoreType
+from ...proto import jina_pb2
+
+
+__all__ = ['NamedScore']
+
+
+class NamedScore:
+
+    def __init__(self, score: Optional[jina_pb2.NamedScoreProto] = None,
+                 copy: bool = False, **kwargs):
+        self._score = jina_pb2.NamedScoreProto()
+        try:
+            if isinstance(score, jina_pb2.NamedScoreProto):
+                if copy:
+                    self._score.CopyFrom(score)
+                else:
+                    self._score = score
```

Comment:
I would implement the full constructor as `Document` as possible, as later those primitive types may inherit from the same base class `jina.dtype`

Reply:
what are you missing? it is almost equal, just the `if self._score is not None`?